src/drivers/distance_sensor/lightware_sf45_serial/lightware_sf45_serial.cpp

```diff
@@ -721,38 +714,76 @@ void SF45LaserSerial::sf45_process_replies(float *distance_m)
 
 				break;
 
-			case 2:
+			case ROTATION_RIGHT_FACING:
 				scaled_yaw = scaled_yaw + 90; // rotation facing right
 				break;
 
-			case 3:
+			case ROTATION_LEFT_FACING:
 				scaled_yaw = scaled_yaw - 90; // rotation facing left
 				break;
 
 			default:
 				break;
 			}
 
-			// Convert to meters for rangefinder update
+			// Convert to meters for the debug message
 			*distance_m = raw_distance * SF45_SCALE_FACTOR;
-			obstacle_dist_cm = (uint16_t)raw_distance;
+			_current_bin_dist = ((uint16_t)raw_distance < _current_bin_dist) ? (uint16_t)raw_distance : _current_bin_dist;
 
 			uint8_t current_bin = sf45_convert_angle(scaled_yaw);
 
-			// If we have moved to a new bin
-
 			if (current_bin != _previous_bin) {
+				PX4_DEBUG("scaled_yaw: \t %d, \t current_bin: \t %d, \t distance: \t %8.4f\n", scaled_yaw, current_bin,
+					  (double)*distance_m);
+
+				if (_current_bin_dist > _obstacle_map_msg.max_distance) {
+					_current_bin_dist = _obstacle_map_msg.max_distance + 1; // As per ObstacleDistance.msg definition
+				}
+
+				// if the sensor has its cycle delay configured for a low value like 5, it can happen that not every bin gets a measurement.
+				// in this case we assume the measurement to be valid for all bins between the previous and the current bin. win
+				uint8_t start;
+				uint8_t end;
+
+				if (abs(current_bin - _previous_bin) > BIN_COUNT / 4) {	// wrap-around case
+					// TODO: differentiate direction of wrap-around
+					start = math::max(_previous_bin, current_bin);
+					end = math::min(_previous_bin, current_bin);
+
+				} else if (_previous_bin < current_bin) {	// Scanning clockwise
+					start = _previous_bin + 1;
+					end = current_bin;
+
+				} else { 						// scanning counter-clockwise
+					start = current_bin;
+					end = _previous_bin - 1;
+				}
+
+				if (start <= end) {
+					for (uint8_t i = start; i <= end; i++) {_obstacle_map_msg.distances[i] = _current_bin_dist;}
```

Comment:
Looks like you covered all the edge cases here. Eventually setting the cycle delay and pps for the lidar can be done in PX4, but getting it working with default settings is probably best. Lightware studio should really be used once to set the scan angle and enable scanning at first, and then not touched again unless debugging.